generator/test_case_generator.go

```diff
@@ -36,69 +36,85 @@ type testConfig struct {
 	// e.g. statsd can have a multiple terraform module sets for difference test scenarios (ecs, eks or ec2)
 	testDir      string
 	terraformDir string
+	// define target matrix field as set(s)
+	// empty map means a testConfig will be created with a test entry for each entry from *_test_matrix.json
+	targets map[string]map[string]struct{}
```

Comment:
So these really seem like filters to decide whether or not an entry in the json should be retained in the matrix.

Why does need to be a map of map? Can this just be a map where the key value pairs match the key value pairs in the json?
So for example, a map with entry <"arc": "arm64">

Reply:
it's map of sets for fast lookup eg
```
{
   "arc": ["amd", "arm"],
   "os": ["al2", "ol9"],
}
```